### DISCUSSION TITLE
Using Set-PnPGroup to set associated group causing permission error

### DISCUSSION THREAD

**Discussion by minassn1** [pnp/powershell#215] [Q&A]
I'm trying to set an associated group as Owners using Set-PnPGroup but getting **"Set-PnPGroup : Access denied. You do not have permission to perform this action or access this resource."**

`Set-PnPGroup -Identity 'Test Owners' -SetAssociatedGroup Owners`

even though I created the new SPO subsite along with the groups and all other configuration using the same account with no issues. Is this a bug?

**Comment by ToddKlindt** [comment]
Which version of the module are you using? 
`Get-Module pnp.powershell -All`

What parameters did you use with `Connect-PnPOnline` when you connected? 
If I remember correctly Set-PnPGroup works when you're a site collection admin and connect to the site itself. Use Set-PnPSiteGroup when you connect to the Admin URL as a Global or SharePoint admin.

**Reply by minassn1** [reply]
1. PnP.PowerShell 1.2.0.0
2. Connect-PnPOnline -URL "https://..." -UseWebLogin
3. I'm connecting to the site itself and I'm a site collection admin, not a global admin on the tenant though.

Could the script executiong settings affect it? I have to ask the tenant admin to check that.

**Reply by ToddKlindt** [reply]
Has anyone run Register-PnPManagementShellAccess and created the Enterprise App Registration in Azure AD?
Also, the post stable builds have added a  -Interactive parameter that should be used instead of -WebLogin. The PnP team put -WebLogin back in kicking and screaming. :)

**Reply by minassn1** [reply]
Thanks I'll check those out.
